scripts/install-foundry.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -e
+
+SCRIPTS_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+MONOMER_DIR=$(cd "$SCRIPTS_DIR/.." && pwd)
+
+# Check for jq installation
+if ! command -v jq &> /dev/null
+then
+    echo "jq could not be found, please install jq for your system at https://jqlang.github.io/jq/download/"
+    exit 1
+fi
+
+# Grab the foundry commit hash.
+SHA=$(jq -r .foundry < "$MONOMER_DIR"/versions.json)
+
+# All code below this comment is copied from the original optimism/ops/scripts/install-foundry.sh
+# script at commit 24a8d3e0
+
+# Check if there is a nightly tag corresponding to the commit hash
+TAG="nightly-$SHA"
+
+# If the foundry repository exists and a branch is checked out, we need to abort
+# any changes inside ~/.foundry/foundry-rs/foundry. This is because foundryup will
+# attempt to pull the latest changes from the remote repository, which will fail
+# if there are any uncommitted changes.
+if [ -d ~/.foundry/foundry-rs/foundry ]; then
+  echo "Foundry repository exists! Aborting any changes..."
+  cd ~/.foundry/foundry-rs/foundry
+  git reset --hard
+  git clean -fd
+  cd -
+fi
+
+# Create a temporary directory
+TMP_DIR=$(mktemp -d)
+echo "Created tempdir @ $TMP_DIR"
+
+# Clone the foundry repo temporarily. We do this to avoid the need for a personal access
+# token to interact with the GitHub REST API, and clean it up after we're done.
+git clone https://github.com/foundry-rs/foundry.git "$TMP_DIR" && cd "$TMP_DIR"
+
+# If the nightly tag exists, we can download the pre-built binaries rather than building
+# from source. Otherwise, clone the repository, check out the commit SHA, and build `forge`,
+# `cast`, `anvil`, and `chisel` from source.
+if git rev-parse "$TAG" >/dev/null 2>&1; then
+  echo "Nightly tag exists! Downloading prebuilt binaries..."
+  foundryup -v "$TAG"
```

Comment:
Have you run this locally? I have a slight preference toward pinning against a version with an available built binary.

I've had a bad time building forge locally, to the point that I've only reliably succeeded with some custom compiler flags.

Reply:
Yeah I tested it locally and was able to build `forge` fine. It took around ~20 mins to compile the projects since it uses `cargo --release` mode but it built fine. Would there be any concerns with us pinning forge for monomer to a different version than forge for optimism?